br/pkg/lightning/errormanager/errormanager.go

```diff
@@ -504,8 +526,129 @@ func (em *ErrorManager) ResolveAllConflictKeys(
 	return errors.Trace(g.Wait())
 }
 
+// ReplaceConflictKeys query all conflicting rows (handle and their
+// values) from the current error report and resolve them concurrently
+// by replacing the necessary rows and reserving the others.
+func (em *ErrorManager) ReplaceConflictKeys(
+	ctx context.Context,
+	tbl tidbtbl.Table,
+	tableName string,
+	pool *utils.WorkerPool,
+	decoder *kv.TableKVDecoder,
+	fnGetLatest func(ctx context.Context, key []byte) ([]byte, error),
+	fnDeleteKey func(ctx context.Context, key []byte) error,
+) error {
+	if em.db == nil {
+		return nil
+	}
+
+	sessionOpts := encode.SessionOptions{
+		// TODO: need to find the correct value for SQLMode
+		SQLMode: mysql.ModeStrictAllTables,
+	}
+	encoder, err := kv.NewBaseKVEncoder(&encode.EncodingConfig{
+		Table:          tbl,
+		SessionOptions: sessionOpts,
+		Logger:         em.logger,
+	})
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	g, gCtx := errgroup.WithContext(ctx)
+	pool.ApplyOnErrorGroup(g, func() error {
+		// check index KV first
```

Comment:
It's better to add more comments to let other reviewers and future developers to understand what we are doing. Maybe a link to https://github.com/lyzx2001/tidb-conflict-replace and also some short comments. In future I think we'd better provide a detailed document to explain the algorithm and link it here.